src/components/Table/Table.Body.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react'
+
+import * as equal from 'fast-deep-equal'
+
+import Row from './Table.Row'
+
+import { BodyProps, BodyState } from './Table.types'
+
+export default class Body extends React.Component<BodyProps, BodyState> {
+  columnsCache = []
+  state: BodyState = { rows: [] }
+
+  constructor(props) {
+    super(props)
+    this.columnsCache = this.getColumnsCache(this.props.columns)
+
+    this.state.rows = this.getRows(props)
+  }
+
+  getRows(props: any = {}) {
+    const { isTableCollapsed, rows, maxRowsToDisplay } = props
+
+    if (!rows) {
+      return []
+    }
+
+    return isTableCollapsed ? rows.slice(0, maxRowsToDisplay) : [...rows]
+  }
+
+  getColumnsCache(columns) {
+    return columns.map(c => {
+      let key = c.columnKey
+      if (Array.isArray(key)) {
+        key = key.join(',')
+      }
+      return key
+    })
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { rows } = this.props
```

Comment:
@plbabin Since we're update state based on `rows`, would it be good to do a comparison between `this.props.rows` vs `nextProps.rows`?

Reply:
we could, but then we will run the deep comparison in both page (here when the props value changed, and then in shouldComponentUpdate). I wanted to reduce those `equal` that as much as possible